Add support for the Edits API

SunburstEnzo opened the pull request:
<!-- Thanks for contributing to MacPaw/OpenAI 😊 -->

## What

• Add "/v1/edits" endpoint
• Some incorrect "File.swift" naming fixes
• "Edits" related tests

## Why

Feature request #48 

## Affected Areas

Enhancement only.

## Notes

• May not need its own version of Choice and Usage structs in EditsResult.
• EditsQuery input may be able to default to "" but OpenAI docs unclear (made it optional, but no default value in init).

Krivoblotsky commented:
Nice, thanks!

What do you think of extending `Model` typealias by adding edits models?
```
model
You can use the text-davinci-edit-001 or code-davinci-edit-001 model with this endpoint.
``

Krivoblotsky commented:
Everything else looks good to me 👍

SunburstEnzo commented:
Oh yeah! Not sure how I missed that, I'll make sure everything listed at https://platform.openai.com/docs/models/model-endpoint-compatibility is added as well 👍

SunburstEnzo commented:
I updated the list and added comments to most 👍

I was thinking about adding a deprecation warning to the temporary models, but left it out for now as it can't really be done for dates, more versions:

`@available(*, deprecated, message: "gpt4_0314 is a temporary snapshot that will be deprecated on June 1st, 2023. If you want to use the latest model version, use gpt-4.")`
`static let gpt4_0314 = "gpt-4-0314"`

<img width="735" alt="Screenshot 2023-04-16 at 5 05 11 pm" src="https://user-images.githubusercontent.com/1451896/232325737-e4617b43-e7c9-4b08-9e9b-a87700af0f36.png">

Btw, I did not add `code-davinci-edit-001` as Codex is already deprecated.

Krivoblotsky commented:
Nice update, thanks, @SunburstEnzo!
